oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/OAuth2AuthorizationRequestRedirectFilter.java

```diff
@@ -112,14 +100,23 @@ public OAuth2AuthorizationRequestRedirectFilter(ClientRegistrationRepository cli
 	 * @param clientRegistrationRepository the repository of client registrations
 	 * @param authorizationRequestBaseUri the base {@code URI} used for authorization requests
 	 */
-	public OAuth2AuthorizationRequestRedirectFilter(
-		ClientRegistrationRepository clientRegistrationRepository, String authorizationRequestBaseUri) {
-
-		Assert.hasText(authorizationRequestBaseUri, "authorizationRequestBaseUri cannot be empty");
+	public OAuth2AuthorizationRequestRedirectFilter(ClientRegistrationRepository clientRegistrationRepository,
+													String authorizationRequestBaseUri) {
 		Assert.notNull(clientRegistrationRepository, "clientRegistrationRepository cannot be null");
-		this.authorizationRequestMatcher = new AntPathRequestMatcher(
-			authorizationRequestBaseUri + "/{" + REGISTRATION_ID_URI_VARIABLE_NAME + "}");
-		this.clientRegistrationRepository = clientRegistrationRepository;
+		Assert.hasText(authorizationRequestBaseUri, "authorizationRequestBaseUri cannot be empty");
+		this.authorizationRequestResolver = new DefaultOAuth2AuthorizationRequestResolver(
+				clientRegistrationRepository, authorizationRequestBaseUri);
+	}
+
+	/**
+	 * Sets the resolver used for resolving {@link OAuth2AuthorizationRequest}'s.
+	 *
+	 * @since 5.1
+	 * @param authorizationRequestResolver the resolver used for resolving authorization requests
+	 */
+	public final void setAuthorizationRequestResolver(OAuth2AuthorizationRequestResolver authorizationRequestResolver) {
```

Comment:
This should be a separate constructor. The reason is that with a custom OAuth2AuthorizationRequestResolver implementation, the ClientRegistrationRepository should not be required for this class. However, it is required to construct the class at the moment.

Reply:
Good catch. Makes sense.